lib/models/measure_source.js

```diff
@@ -0,0 +1,74 @@
+const MeasureSchema = require('cqm-models').MeasureSchema;
+const mongoose = require('mongoose');
+
+module.exports = class MongoDBMeasureSource {
+  constructor(connectionInfo) {
+    mongoose.connect(connectionInfo);
+
+    this.CQLMeasure = mongoose.model('Measure', MeasureSchema);
+    this.measure = null;
+  }
+
+  /**
+   * This method searches for a measure given its ObjectId; if it finds it, it sets the measure property and runs callbacks, if any
+   * @param { string | mongoose.Types.ObjectId } measureId - The measure ObjectId (unique MongoDB identifier) to search by. If a string is passed in, it will be converted to an ObjectId
+   * @param {function} callback - A function to run after searching for the measure
+   * @returns {Measure | Error } - the Measure found, or an Error
+   */
+  findMeasure(measureId, callback = null) {
+    const self = this;
+    let measureIdObj = null;
+    let err = null;
+
+    if (typeof measureId === 'string') {
+      measureIdObj = mongoose.Types.ObjectId(measureId);
+    } else if (measureId instanceof mongoose.Types.ObjectId) {
+      measureIdObj = measureId;
+    }
+
+    if (measureIdObj === null) return TypeError('measureId must be string or ObjectId');
+
+    [err, this.measure] = this.CQLMeasure.findById(measureIdObj);
```

Comment:
This (and in `findMeasureByUser`) is not correct syntax. You'll get back a query but it won't execute unless you pass a callback to findById, and in that object `this.measure` is set to the retrieved measure.